Add support for more OTel exporter environment variables and Configuration

antontroshin opened the pull request:
# Description

<!--
Please explain the changes you've made.
-->
New environment variables and configuration fields to set up for OTel exporter
Env variables:
- OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
- OTEL_EXPORTER_OTLP_TRACES_PROTOCOL
- OTEL_EXPORTER_OTLP_HEADERS
- OTEL_EXPORTER_OTLP_TRACES_HEADERS
- OTEL_EXPORTER_OTLP_TIMEOUT
- OTEL_EXPORTER_OTLP_TRACES_TIMEOUT

## Issue reference

<!--
We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.
-->

Please reference the issue this PR will close: #8143 
## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [ ] Code compiles correctly
- [ ] Created/updated tests
- [ ] Unit tests passing
- [ ] End-to-end tests passing
- [ ] Extended the documentation / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_
- [ ] Specification has been updated / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_
- [ ] Provided sample for the feature / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_

mikeee commented on pkg/config/configuration_test.go:
```diff
@@ -479,29 +492,69 @@ func TestSetTracingSpecFromEnv(t *testing.T) {
+	t.Setenv(env.OtlpExporterTracesTimeout, "invalid")
+	conf := LoadDefaultConfiguration()
+	err := SetTracingSpecFromEnv(conf)
+	require.NoError(t, err)
```
```suggestion
	require.Error(t, err)
```
This test case looks like it should always error with an invalid string to int conversion?

antontroshin replied:
Yes, correct, fixed. Thanks.

mikeee reviewed (commented):
Looking good, just one test case I noticed will fail.

This appears to encompass proposed changes in #8159, is this PR intended to usurp that one or is it complementary?

antontroshin commented:
> This appears to encompass proposed changes in #8159, is this PR intended to usurp that one or is it complementary?

It seems we were both working on the same thing (unaware of each other) and neither was assigned to the issue. 
In my PR I've added support for additional `TRACES` specific env vars and added a few others. Includes support for standalone and k8s config, and I'm working now on adding `secretKeyRef` expanding from secrets for the headers as they may (and likely) contain secrets or auth values.
